ls1intum/ArTEMiS
# New complaint list page

### Checklist

### Motivation and Context



This pull request creates a new component for listing complaints, that it can be used by tutors to see complaints about themselves, or by instructors to see complaints in a course / exercise.

This implements [ARTEMIS-348](https://jirabruegge.in.tum.de/browse/ARTEMIS-348) and [ARTEMIS-349](https://jirabruegge.in.tum.de/browse/ARTEMIS-349)

### Description

I introduced a new complaint service to manage all the different logic about complaints, and I have also rewritten the complaint resource to use only the complaint service and not the complaint repository directly to guarantee proper encapsulation and separation of concerns.

I have also noticed that we weren't actually calculating the number of complaints about a single tutor in the course dashboard, so I fixed that as well.

### Steps for Testing


As a tutor, you can access to complaints about you in two ways:
- From a tutor course dashboard, you can click on "Complaints about your assessments" and you will see complaints about you for that course
- From a tutor exercise dashboard, you can click on "Complaints about your assessments" and you will see complaints about you for that exercise

As an instructor, there are different ways:
-  From the instructor course dashboard, you click on "complaints" to see all the complaints in the course
-  From the instructor course dashboard, you go to the tutor leaderboard and click on the number of complaints near to a tutor, to see all the complaints about that tutor in the course
-  From the instructor exercise dashboard, you click on "complaints" to see all the complaints in the exercise
-  From the instructor exercise dashboard, you go to the tutor leaderboard and click on the number of complaints near to a tutor, to see all the complaints about that tutor in the exercise

### Screenshots

![image](https://user-images.githubusercontent.com/1713343/58368368-6b5ab880-7eec-11e9-9f01-825974427c38.png)
